Source/VolunteerReporting/Web/Controllers/CaseReportsController.cs

```diff
@@ -27,21 +27,32 @@ public CaseReportsController(ICaseReports caseReports, IHealthRisks healthRisks,
 
         [HttpGet]
         public async Task<IEnumerable<CaseReportExpanded>> Get()
-        {
+        {   
             var caseReports = await _caseReports.GetAllAsync();
 
+            // Comment from woksin - 15/02-2018
+            // By fetching all risks and dataCollectors to memory we should get reduced latency,
+            // which was the case when I tested this method using the old method versus fetching the data prior to querying it.
+            // In my opinion, the best way to do this is to have a cache-system for these databases (preferably in the classes that deals directly
+            // with IMongoDatabase and IMongoCollection 
+            var healthRisks = await _healthRisks.getAllAsync();
```

Comment:
This will work untill data collectors  gets to large, but since they will be stored per national society it might work for the forceable future. Healthrisks should always work :)

Reply:
Do you mean it might be a problem considering memory?

Reply:
There might a need to revisit this whole controller and its models. 
The `CaseReportExpanded` has a very non descriptive name - when looking at it it is really just a projection as well. We could probably do this a bit smarter - **no caching** though. With the events we could have another MongoDB collection that held all this as a projection, rather than having to query all the different ones to get this. 

I do recall us talking a bit about it as a temporary solution...

Reply:
Btw: Nitpicking; C# convention is UpperCamelCase :) - getAllAsync() -> GetAllAsync()

Reply:
@einari Woops.. Java guy here...

Reply:
I will fix that asap

Reply:
Good point @einari. Did not think about that. Thats kinda the whole idea behind event sourcing and read models :)

Reply:
@einari I agree. I'll just let it be as-is for the moment then.